rom/dev/src/flow/cold_reset/fw_processor.rs

```diff
@@ -542,48 +548,28 @@ impl FirmwareProcessor {
     ///
     /// # Arguments
     ///
-    /// * `env`  - ROM Environment
     /// * `info` - Image Verification Info
+    /// * `persistent_data` - Persistent data accessor
+    ///
     #[cfg_attr(not(feature = "no-cfi"), cfi_impl_fn)]
     fn populate_data_vault(
-        data_vault: &mut DataVault,
         info: &ImageVerificationInfo,
-        persistent_data: &PersistentDataAccessor,
+        persistent_data: &mut PersistentDataAccessor,
     ) {
-        data_vault.write_cold_reset_entry48(ColdResetEntry48::FmcTci, &info.fmc.digest.into());
+        let manifest_address = &persistent_data.get().manifest1 as *const _ as u32;
+        let data_vault = &mut persistent_data.get_mut().data_vault;
+        data_vault.set_fmc_tci(&info.fmc.digest.into());
+        data_vault.set_fmc_svn(info.fw_svn);
+        data_vault.set_fmc_entry_point(info.fmc.entry_point);
+        data_vault.set_owner_pk_hash(&info.owner_pub_keys_digest.into());
+        data_vault.set_vendor_ecc_pk_index(info.vendor_ecc_pub_key_idx);
+        data_vault.set_vendor_pqc_pk_index(info.vendor_pqc_pub_key_idx);
+        data_vault.set_rt_tci(&info.runtime.digest.into());
+        data_vault.set_rt_svn(info.fw_svn);
+        data_vault.set_rt_min_svn(info.fw_svn);
+        data_vault.set_rt_entry_point(info.runtime.entry_point);
+        data_vault.set_manifest_addr(manifest_address);
```

Comment:
Should we be setting PMP protections here as well?

Reply:
Yes, that will come later though; I mentioned that in the commit. From the commit message "Memory protections on the DCCM datavault will be added as a separate change later."